src/components/Card.js

```diff
@@ -1,41 +1,99 @@
+import Api from "../components/Api.js";
+import DeleteCardForm from "./DeleteCardForm.js";
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "b685d3e0-616a-4dae-bc5b-53892a4f7953",
+    "Content-Type": "application/json",
+  },
+});
+
 export default class Card {
-  constructor(cardData, cardSelector, handleCardClick) {
+  constructor(
+    cardData,
+    cardSelector,
+    handleCardClick,
+    handleCardDeleteFunctionInIndexComponent
+  ) {
     this._cardData = cardData;
+    this._id = cardData._id;
     this._name = cardData.name;
     this._link = cardData.link;
+    this._owner = cardData.owner._id;
+    this._cardId = cardData._id;
+    this._likes = cardData.likes;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleCardDeleteFunctionInIndexComponent =
+      handleCardDeleteFunctionInIndexComponent;
   }
 
-  _handleFavIcon = () => {
-    this._favIconElement.classList.toggle("card__fav-icon-selected");
+  _handleFavIconClick = () => {
+    //variable to register if current user liked the image or not
+    const currentUserLike = this._likes.find((user) => {
+      return user._id === "f50447686616d1fa985ca0e1";
+    });
+    let didCurrentUserLikeThisCard = currentUserLike ? true : false;
+    if (didCurrentUserLikeThisCard) {
+      // 1. Make a call to network to remove the users like for this card
+      api.unLikeACard(this._cardId).then((data) => {
+        this._likes = data.likes;
+        this._favIconElement.classList.remove("card__fav-icon-selected");
+        this._favCountElement.textContent = this._likes.length;
+      });
+    } else {
+      // 1. Make a call to network to add the users like for this card
+      api.likeACard(this._cardId).then((data) => {
+        this._likes = data.likes;
+        // 2. Update state of the like button and update counter
+        this._favIconElement.classList.toggle("card__fav-icon-selected");
+        this._favCountElement.textContent = this._likes.length;
+      });
+    }
+  };
+
+  _handleDeleteCardFunctionInCardComponent = () => {
+    this.deleteCardPopup = new DeleteCardForm(
+      "#delete-image-confirm-modal",
+      this._handleDeleteCardFormSubmit
+    );
+    this.deleteCardPopup.openModal();
   };
 
-  _handleDelButton = () => {
+  _handleDeleteCardFormSubmit = () => {
     this._cardElement.remove();
-    this._cardElement = null;
+    this._handleCardDeleteFunctionInIndexComponent(this._cardId);
+    if (this.deleteCardPopup) {
+      this.deleteCardPopup.closeModal();
+    }
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well